src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,719 @@
+                    updateEstimate = true;
+            }
+
+            long[] deltas = new long[]{tsDelta, tDelta, sizeDelta};
+            if (interArrival.computeDeltas(timestamp, arrivalTimeMs,
+                    (int) payloadSize, deltas, nowMs))
+            {
+                double tsDeltaMs = (1000.0 * tsDelta)
+                        / (1 << kInterArrivalShift);
+                estimator.update(tDelta, tsDeltaMs, sizeDelta,
+                        detector.getState());
+                detector.detect(estimator.getOffset(), tsDeltaMs,
+                        estimator.getNumOfDeltas(), arrivalTimeMs);
+            }
+
+            if (!updateEstimate)
+            {
+                // Check if it's time for a periodic update or if we
+                // should update because of an over-use.
+                if (lastUpdateMs == -1 ||
+                        nowMs - lastUpdateMs > remoteRate
+                                .getFeedBackInterval())
+                {
+                    updateEstimate = true;
+                }
+                else if (detector.getState() == BandwidthUsage.kBwOverusing)
+                {
+                    long incomingRate_ =
+                            incomingBitrate.getRate(arrivalTimeMs);
+                    if (incomingRate_ > 0 &&
+                            remoteRate.isTimeToReduceFurther(nowMs,
+                                    incomingBitrate_))
+                    {
+                        updateEstimate = true;
+                    }
+                }
+            }
+            if (updateEstimate)
+            {
+                // The first overuse should immediately trigger a new estimate.
+                // We also have to update the estimate immediately if we are
+                // overusing and the target bitrate is too high compared to
+                // what we are receiving.
+                RateControlInput input = new RateControlInput(detector
+                        .getState(), incomingBitrate.getRate(arrivalTimeMs),
+                        estimator.getVarNoise());
+
+                /**
+                 * @Todo remote.update returns nothing. This is fine since we
+                 * can retrieve the value from remoteRate.getLatestEstimate
+                 */
+                remoteRate.update(input, nowMs);
+                targetBitrateBps = remoteRate.getLatestEstimate();
+                updateEstimate = remoteRate.isValidEstimate();
+            }
+        }
+        if (updateEstimate)
+        {
+            lastUpdateMs = nowMs;
+            observer_.onReceiveBitrateChanged(getSsrcs(), targetBitrateBps);
+        }
+    }
+
+    void timeoutStreams(long nowMs)
+    {
+        synchronized (critSect) {
+            Iterator<Map.Entry<Long, Long>> itr = ssrcs_.entrySet().iterator();
+            while (itr.hasNext()) {
+                /**
+                 * @Todo ask if this is the best way to modify a treemap object
```

Comment:
Yeah, this looks like the right way to modify a map. You can avoid the cast below if you declare "entry" to be of type Map.Entry<Long, Long>